src/p5.dance.js

```diff
@@ -134,6 +138,68 @@ module.exports = class DanceParty {
     this.p5_.remove();
   }
 
+  loadSprites(animationData, spriteNames) {
+    const promises = [];
+    spriteNames.forEach(costume => {
+      if (this.animations[costume].length === this.world.MOVE_NAMES.length) {
+        // Already loaded, nothing to do:
+        return;
+      }
+      // We have found a new sprite to load, reset allSpritesLoaded to false
+      this.allSpritesLoaded = false;
+      const costumeData = animationData[costume.toLowerCase()];
+      this.world.MOVE_NAMES.forEach(({ name: moveName, mirror }, moveIndex) => {
+        const moveData = costumeData[moveName.toLowerCase()];
+
+        promises.push(new Promise(resolveSpriteSheet => {
+          // Passing a callback as the 3rd arg to loadSpriteSheet() indicates that
+          // we want it to load the image as a Image (instead of a p5.Image), which
+          // avoids a canvas creation. This makes it possible to run on mobile
+          // Safari in iOS 12 with canvas memory limits.
+          const spriteSheet = this.resourceLoader_.loadSpriteSheet(
+            moveData.spritesheet,
+            moveData.frames,
+            () => {
+              const animation = this.p5_.loadAnimation(spriteSheet);
+              this.setAnimationSpriteSheet(
+                costume,
+                moveIndex,
+                spriteSheet,
+                mirror,
+                animation
+              );
+              resolveSpriteSheet();
+            }
+          );
+        }));
+      });
+    });
+    const promise = Promise.all(promises);
+    promise.then(() => {
+      this.allSpritesLoaded = true;
+    });
+    return promise;
+  }
+
+  async ensureSpritesAreLoaded(spriteNames) {
+    spriteNames = spriteNames || this.world.SPRITE_NAMES;
```

Comment:
Nit: Give the parameter a default.

Reply:
I’d like the default to be undefined. This way, older tests can still call it and get the full
set of moves. Would you prefer to see it as
`async ensureSpritesAreLoaded(spriteNames = undefined)` ?

Reply:
I'm saying this:
```js
  async ensureSpritesAreLoaded(spriteNames) {
    spriteNames = spriteNames || this.world.SPRITE_NAMES;
```
...could probably just be this?
```js
  async ensureSpritesAreLoaded(spriteNames = this.world.SPRITE_NAMES) {
```